lib/connect/index.js

```diff
@@ -1,146 +1,153 @@
-
-    if (opts.protocol === null) {
-      throw new Error('Missing protocol')
-    }
-    opts.protocol = opts.protocol.replace(/:$/, '')
-  }
-
-  // merge in the auth options if supplied
-  parseAuthOptions(opts)
-
-  // support clientId passed in the query string of the url
-  if (opts.query && typeof opts.query.clientId === 'string') {
-    opts.clientId = opts.query.clientId
-  }
-
-  if (opts.cert && opts.key) {
-    if (opts.protocol) {
-      if (['mqtts', 'wss'].indexOf(opts.protocol) === -1) {
-        /*
-         * jshint and eslint
-         * complains that break from default cannot be reached after throw
-         * it is a foced exit from a control structure
-         * maybe add a check after switch to see if it went through default
-         * and then throw the error
-        */
-        /* jshint -W027 */
-        /* eslint no-unreachable:1 */
-        switch (opts.protocol) {
-          case 'mqtt':
-            opts.protocol = 'mqtts'
-            break
-          case 'ws':
-            opts.protocol = 'wss'
-            break
-          default:
-            throw new Error('Unknown protocol for secure connection: "' + opts.protocol + '"!')
-            break
-        }
-        /* eslint no-unreachable:0 */
-        /* jshint +W027 */
-      }
-    } else {
-      // don't know what protocol he want to use, mqtts or wss
-      throw new Error('Missing secure protocol key')
-    }
-  }
-
-  if (!protocols[opts.protocol]) {
-    var isSecure = ['mqtts', 'wss'].indexOf(opts.protocol) !== -1
-    opts.protocol = [
-      'mqtt',
-      'mqtts',
-      'ws',
-      'wss'
-    ].filter(function (key, index) {
-      if (isSecure && index % 2 === 0) {
-        // Skip insecure protocols when requesting a secure one.
-        return false
-      }
-      return (typeof protocols[key] === 'function')
-    })[0]
-  }
-
-  if (opts.clean === false && !opts.clientId) {
-    throw new Error('Missing clientId for unclean clients')
-  }
-
-  function wrapper (client) {
-    if (opts.servers) {
-      if (!client._reconnectCount || client._reconnectCount === opts.servers.length) {
-        client._reconnectCount = 0
-      }
-
-      opts.host = opts.servers[client._reconnectCount].host
-      opts.port = opts.servers[client._reconnectCount].port
-      opts.hostname = opts.host
-
-      client._reconnectCount++
-    }
-
-    return protocols[opts.protocol](client, opts)
-  }
-
-  return new MqttClient(wrapper, opts)
-}
-
-module.exports = connect
-module.exports.connect = connect
-module.exports.MqttClient = MqttClient
+'use strict'
+
+var MqttClient = require('../client')
+var url = require('url')
+var xtend = require('xtend')
+var protocols = {}
+
+if (process.title !== 'browser') {
+  protocols.mqtt = require('./tcp')
+  protocols.tcp = require('./tcp')
+  protocols.ssl = require('./tls')
+  protocols.tls = require('./tls')
+  protocols.mqtts = require('./tls')
+}
+
+protocols.wx = require('./wx')
+protocols.wxs = require('./wx')

```

Comment:
we should not be adding `wx` if it's not a browser. the main `wx` object is not there.